lib/benchee/formatters/csv.ex

```diff
@@ -57,19 +59,30 @@ defmodule Benchee.Formatters.CSV do
       ...> }
       iex> suite
       iex> |> Benchee.Formatters.CSV.format
+      iex> |> Enum.fetch!(0)
       iex> |> (fn({rows, _}) -> Enum.take(rows, 2) end).()
       ["Name,Input,Iterations per Second,Average,Standard Deviation,Standard Deviation Iterations Per Second,Standard Deviation Ratio,Median,Minimum,Maximum,Sample Size\\r\\n",
        "My Job,Some Input,2.0e3,500.0,200.0,800.0,0.4,450.0,200,900,8\\r\\n"]
 
   """
-  @spec format(Suite.t) :: {Enumerable.t, String.t}
+  @spec format(Suite.t) :: [{Enumerable.t, String.t}]
   def format(%Suite{scenarios: scenarios, configuration: configuration}) do
-    rows = scenarios
-           |> Enum.sort_by(fn(scenario) -> scenario.input_name end)
-           |> Enum.map(&to_csv/1)
-           |> add_headers
+    sorted_scenarios = Enum.sort_by(scenarios, fn(scenario) -> scenario.input_name end)
+
+    [
+      {get_general_benchmarks(sorted_scenarios), get_filename(configuration)},
+      {get_raw_benchmarks(sorted_scenarios), @raw_benchmark_output}
+    ]
+  end
+
+  defp get_general_benchmarks(scenarios), do: get_benchmarks(scenarios, &General.to_csv/1, &General.add_headers/1)
+  defp get_raw_benchmarks(scenarios), do: get_benchmarks(scenarios, &Raw.to_csv/1, &Raw.add_headers/1)
+
+  defp get_benchmarks(scenarios, to_csv, add_headers) do
+    scenarios
+           |> Enum.map(to_csv)
+           |> add_headers.()
            |> CSV.encode()
```

Comment:
formatting looks off here, I think it should be:

```
scenarios
|> fun
|> fun
```

Reply:
ok